### PR TITLE
KAFKA-18050: Upgrade the checkstyle version to 10.20.2

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-18050 

As we don't support the Java version8, we can avoid the CVE-2023-2976 and CVE-2020-8908m thus we should upgrade the checkstyle version, and some file are missed in old version checkstyle, thus we should upgrade the checkstyle version.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dejan2609** [comment]
Related PR: #10967

@m1a2st I will try to test my PR on top of your changes.

**Comment by chia7712** [comment]
@m1a2st please rebase code

**Comment by dejan2609** [comment]
@m1a2st you may also use  latest Checkstyle patch version (10.20.2)

**Comment by m1a2st** [comment]
@dejan2609 I found there is an error when I use the version of 10.20.2, PTAL, Thanks for your helping https://github.com/apache/kafka/actions/runs/12122917186/job/33797238123?pr=17999

**Comment by dejan2609** [comment]
No problem @m1a2st  ! Upgrade to version 10.20.1 is still awesome.

You may want to squash commits into one  so maintainer (@chia7712) could merge your change into the trunk.

**Comment by romani** [comment]
It will be awesome to create separate PR after merge of this PR to update to latest checkstyle and share report content to let us review how it was possible to generate bad xml

**Comment by dejan2609** [comment]
> It will be awesome to create separate PR after merge of this PR to update to latest checkstyle and share report content to let us review how it was possible to generate bad xml

I agree with that, @romani. 

Anyhow: I tested (on my computer) this branch with Checkstyle 10.20.2 and it compiles just fine. 
Gradle command used: `./gradlew checkstyleMain checkstyleTest -x test --info`

Maybe Github Actions build went berserk :smiley: ? We will check eventually.

**Comment by romani** [comment]
Let's merge this PR first.

**Comment by chia7712** [comment]
> I tested (on my computer) this branch with Checkstyle 10.20.2 and it compiles just fine.

I run the 10.20.2 on #18010, and it works well.

> Let's merge this PR first.

I have updated this PR to use version 10.20.2 again. Let's see what happens this time.

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -61,7 +61,7 @@ versions += [
   // when updating checkstyle, check whether the exclusion of
```
Comment (chia7712) [inline gradle/dependencies.gradle]: Could you please remove those comments? they are out-of-date after you upgrade the checkstyle version

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for this patch. one small comment is left.

**Review by chia7712 — APPROVED** [review]

**Comment by chia7712** [comment]
The failed test is traced by https://issues.apache.org/jira/browse/KAFKA-18036

**Pull request merged**